sklearn/metrics/_plot/precision_recall_curve.py

```diff
@@ -186,21 +310,42 @@ def plot(
         `drawstyle="default"`. However, the curve will not be strictly
         consistent with the reported average precision.
         """
-        self.ax_, self.figure_, name = self._validate_plot_params(ax=ax, name=name)
-
-        default_line_kwargs = {"drawstyle": "steps-post"}
-        if self.average_precision is not None and name is not None:
-            default_line_kwargs["label"] = (
-                f"{name} (AP = {self.average_precision:0.2f})"
+        names = _deprecate_singular(name, names, "name")
+        names_ = names if names else self.names
+        multi_args = [
+            input
+            for input in (self.average_precisions, names_)
+            if isinstance(input, list)
+        ]
+        multi_args.extend([self.precisions, self.recalls])
+        # I could make this message more informative (i.e. give
+        # lengths of each param) but it requires a fair bit more code
+        if len({len(arg) for arg in multi_args}) > 1:
+            raise ValueError(
+                "`self.precisions`, `self.recalls`, and if provided, "
+                "`self.average_precisions` and `self.names` (or `plot`'s `names`), "
+                "from `PrecisionRecallDisplay` initialization, should all "
+                "be lists of the same length."
```

Comment:
This message could be more informative (i.e. we could give lengths of each param) but that would require a fair bit more code